core/src/components/toggle/toggle.tsx

```diff
@@ -312,7 +343,8 @@ export class Toggle implements ComponentInterface {
           [`toggle-${rtl}`]: true,
         })}
       >
-        <label class="toggle-wrapper">
+        {this.renderFallbackHapticElements()}
+        <label class="toggle-wrapper" onClick={this.onClick}>
```

Comment:
`onClick listener` moved to label, without this change `Host` was capturing clicks on the haptic `<label>` too, not only on `.toggle-wrapper`